torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -1189,18 +1193,19 @@ def make_launcher(self) -> LauncherType:
                 *call_args,
             ]
 
-        exec(
-            f"""
-            def launcher({', '.join(def_args)}, grid, stream):
-                if callable(grid):
-                    grid_0, grid_1, grid_2 = grid(grid_meta)
-                else:
-                    grid_0, grid_1, grid_2 = grid
-                runner({', '.join(runner_args)})
-                return bin
-            """.lstrip(),
-            scope,
-        )
+        if "extra_launcher_args" in self.inductor_meta:
+            def_args.extend(self.inductor_meta["extra_launcher_args"])
+
+        grid = GridExpr.from_meta(self.inductor_meta, cfg)
+        lines = [
+            f"def launcher({', '.join(def_args)}, stream):",
+            *[f"    {line}" for line in grid.prefix],
+            f"    grid_0 = {grid.x_grid}",
+            f"    grid_1 = {grid.y_grid}",
+            f"    grid_2 = {grid.z_grid}",
```

Comment:
To confirm my understanding here, if we passed in a Grid2D or Grid1D based on the inductor meta,  y_grid and z_grid will default to 1 here, correct?

Reply:
Yes correct, they will be 1.